packages/flutter_tools/test/general.shard/crash_reporting_test.dart

```diff
@@ -13,26 +13,30 @@ import 'package:flutter_tools/src/base/platform.dart';
 import 'package:flutter_tools/src/doctor.dart';
 import 'package:flutter_tools/src/project.dart';
 import 'package:flutter_tools/src/reporting/crash_reporting.dart';
-import 'package:flutter_tools/src/reporting/reporting.dart';
 import 'package:http/http.dart';
 import 'package:http/testing.dart';
 import 'package:test/fake.dart';
+import 'package:unified_analytics/unified_analytics.dart';
 
 import '../src/common.dart';
 import '../src/fake_process_manager.dart';
+import '../src/fakes.dart';
 
 void main() {
   late BufferLogger logger;
-  late FileSystem fs;
-  late TestUsage testUsage;
+  late MemoryFileSystem fs;
   late Platform platform;
   late OperatingSystemUtils operatingSystemUtils;
   late StackTrace stackTrace;
+  late FakeAnalytics fakeAnalytics;
 
   setUp(() async {
     logger = BufferLogger.test();
     fs = MemoryFileSystem.test();
-    testUsage = TestUsage();
+    fakeAnalytics = getInitializedFakeAnalyticsInstance(
```

Comment:
What about instead we just use our own implementation of this class, such as:

```dart
class FakeAnalytics extends Fake implements Analytics {
  const FakeAnalytics({this.okToSend = true});
  
  @override
  final bool okToSend;
}
```

Reply:
I haven't done a deep dive on `getInitializedFakeAnalyticsInstance`, but I am inclined to agree with this suggestion. Speaking generally, I think it's good for a test to fake as little as needed so that it's more clear as to what that test depends upon.